src/partitionedNLPModels/utils.jl

```diff
@@ -98,8 +98,58 @@ function compiled_grad_element_function(
   return compiled_f_tape
 end
 
+function merge_linear_elements(vec_element_function::Vector{ExpressionTreeForge.Type_expr_tree}, N::Int)
+  type_element_functions = ExpressionTreeForge.get_type_tree.(vec_element_function)
+  linears = (elt_fun -> ExpressionTreeForge.is_linear(elt_fun) || ExpressionTreeForge.is_constant(elt_fun)).(type_element_functions)
+  indices_linear_elements = filter(i -> linears[i], 1:N)  
+  if !isempty(indices_linear_elements)
+    gathering_linear_elements = ExpressionTreeForge.sum_expr_trees(vec_element_function[indices_linear_elements])
+    indices_nonlinear_elements = filter(i -> !linears[i], 1:N)
+    vec_element_function = vcat(vec_element_function[indices_nonlinear_elements], gathering_linear_elements)
+    N = length(vec_element_function)
+    linear_vector = zeros(Bool, N) # every element function except the last one is nonlinear
+    linear_vector[N] = true # last element function is the only one linear
+  else
+    linear_vector = zeros(Bool, N) # every element function is nonlinear
+  end
+  return vec_element_function, N, linear_vector
+end 
```

Comment:
It might be cleaner to declare this as 2 functions as we are used to in JSO:
(this should make improving the code later easier)
```
function merge_linear_elements(vec_element_function::Vector{ExpressionTreeForge.Type_expr_tree}, N::Int)
  linear_vector = zeros(Bool, N)
  return merge_linear_elements!(vec_element_function, N, linear_vector)
end
```
and then
```
function merge_linear_elements!(vec_element_function::Vector{ExpressionTreeForge.Type_expr_tree}, N::Int, linear_vector::Vector{Bool})
  type_element_functions = ExpressionTreeForge.get_type_tree.(vec_element_function)
  linears = (elt_fun -> ExpressionTreeForge.is_linear(elt_fun) || ExpressionTreeForge.is_constant(elt_fun)).(type_element_functions)
  indices_linear_elements = filter(i -> linears[i], 1:N)  
  if !isempty(indices_linear_elements)
    gathering_linear_elements = ExpressionTreeForge.sum_expr_trees(vec_element_function[indices_linear_elements])
    indices_nonlinear_elements = filter(i -> !linears[i], 1:N)
    vec_element_function = vcat(vec_element_function[indices_nonlinear_elements], gathering_linear_elements)
    N = length(vec_element_function)
    linear_vector .= zero(Bool) # every element function except the last one is nonlinear
    linear_vector[N] = true # last element function is the only one linear
  else
    linear_vector .= zero(Bool) # every element function is nonlinear
  end
  return vec_element_function, N, linear_vector
end 
```

Reply:
Sorry, I missed than `N` changes.

Reply:
No problem, thank you for the review!